Carlos/Composed.swift

```diff
@@ -49,8 +49,23 @@ extension CacheLevel {
         return request.future
       },
       setClosure: { (value, key) in
+        let promise = Promise<()>()
+
         self.set(value, forKey: key)
-        cache.set(value, forKey: key)
+          .onSuccess { _ in
```

Comment:
this diff is the only one I'm not sure about w.r.t. the logic.
Basically we call `cache.set` in both cases (success/failure of the `self.set` call), but if `self.set` succeeds and `cache.set` fails we don't call `promise.fail`, that could be an error. On line `63` we could use `mimic`, but anyway as I said I would revisit this snippet, maybe we can call `self.set` and `cache.set` independently, collect the two resulting `Future`s, and combine them somehow, then `mimic` our `Promise` on the result of the combination of the two `Future`s. 

What do you think?